src/client/NetworkingClient.cpp

```diff
@@ -53,6 +53,11 @@ void NetworkingClient::setDirection(sf::Vector2f direction) {
     mDirection.set(direction);
 }
 
+void NetworkingClient::setGroupable(bool groupable) {
+    mNeedsGroupStateSync = mGroupable != groupable;
```

Comment:
what's this variable for?

Reply:
the api client needs to know when to make an API request to the server, so that it doesn't spam the server. It basically should only make a request when there's a state transition (mGroupable != groupable), that is what's captured here

Reply:
Make sense. Please add a comment explaining this